From a58285bfb1e37b87c7a85afaead9c9496bc1badb Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Sun, 3 Oct 2010 15:58:12 -0400
Subject: [PATCH] fallocate: fix build failure with old linux headers

If linux/falloc.h does not exist, the build system still enables the
fallocate util, but ultimately fails when it tries to include the
header and use a define from it.

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
---
 sys-utils/fallocate.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/sys-utils/fallocate.c b/sys-utils/fallocate.c
index d1f43c2..fd4d2a9 100644
--- a/sys-utils/fallocate.c
+++ b/sys-utils/fallocate.c
@@ -37,7 +37,11 @@
 # include <sys/syscall.h>
 #endif
 
-#include <linux/falloc.h>	/* for FALLOC_FL_* flags */
+#ifdef HAVE_LINUX_FALLOC_H
+# include <linux/falloc.h>	/* for FALLOC_FL_* flags */
+#else
+# define FALLOC_FL_KEEP_SIZE 1
+#endif
 
 #include "nls.h"
 #include "strtosize.h"
-- 
1.7.3

